src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineRuntimeMacros.java

```diff
@@ -0,0 +1,295 @@
+
+import java.util.List;
+
+public class PipelineRuntimeMacros {
+
+  @When("Upload pipeline from file {string}")
+  public void uploadPipelineFromFile(String filename) {
+    Helper.uploadPipelineFromFile(filename);
+  }
+
+  @When("Enter preview mode")
+  public void enterPreviewMode() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("pipeline-preview-btn"));
+  }
+
+  @When("Start preview run")
+  public void startPreviewRun() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("preview-top-run-btn"));
+  }
+
+  @Then("Preview runtime arguments dialog is shown")
+  public void runtimeArgumentsDialogIsShown() {
+    Assert.assertTrue(ElementHelper.isElementDisplayed(
+      By.cssSelector(Helper.getCssSelectorByDataTestId("runtimeargs-preview")),
+      2));
+  }
+
+  @Then("Preview runtime argument row {string} key is disabled")
+  public void runtimeArgumentRowIntKeyIsDisabled(String index) {
+    WebElement rowKeyElement = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+      Helper.getCssSelectorByDataTestId(index) + " " +
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_KEY_SELECTOR) +  " input"
+    );
+    Assert.assertFalse(rowKeyElement.isEnabled());
+  }
+
+  @Then("Preview runtime argument row {string} key field is {string}")
+  public void runtimeArgumentRowIntKeyFieldIsString(String index, String fieldValue) {
+    WebElement keyFieldElement = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+      Helper.getCssSelectorByDataTestId(index) + " " +
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_KEY_SELECTOR) + " input"
+    );
+    Assert.assertEquals(fieldValue, ElementHelper.getElementAttribute(keyFieldElement, "value"));
+  }
+
+  @Then("Preview runtime argument row {string} value field is {string}")
+  public void runtimeArgumentRowIntValueFieldIsString(String index, String fieldValue) {
+    WebElement valueFieldElement = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+      Helper.getCssSelectorByDataTestId(index) + " " +
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_VALUE_SELECTOR) + " textarea"
+    );
+    Assert.assertEquals(fieldValue, ElementHelper.getElementAttribute(valueFieldElement, "value"));
+  }
+
+  @Then("Preview runtime argument row {string} does not exist")
+  public void runtimeArgumentRowIntDoesNotExist(String index) {
+    try {
+      WebElement rowEl = Helper.locateElementByCssSelector(
+        Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+        Helper.getCssSelectorByDataTestId(index) + " " +
+        Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_KEY_SELECTOR));
+      Assert.assertFalse(ElementHelper.isElementDisplayed(rowEl));
```

Comment:
I like `Assert.fail` more here

Reply:
I think in this case, `Assert.assertFail` is more clear and can track that the assertion passed